crates/wasm-mutate/src/lib.rs

```diff
@@ -287,19 +282,65 @@ impl<'wasm> WasmMutate<'wasm> {
                 RemoveItemMutator(Item::Data),
                 RemoveItemMutator(Item::Element),
                 RemoveItemMutator(Item::Tag),
+                ModifyDataMutator,
             )
         );
 
         Err(Error::no_mutations_applicable())
     }
 
+    fn setup(&mut self, input_wasm: &'wasm [u8]) -> Result<()> {
+        self.info = Some(ModuleInfo::new(input_wasm)?);
+        self.rng = Some(SmallRng::seed_from_u64(self.seed));
+        Ok(())
+    }
+
     pub(crate) fn rng(&mut self) -> &mut SmallRng {
         self.rng.as_mut().unwrap()
     }
 
     pub(crate) fn info(&self) -> &ModuleInfo<'wasm> {
         self.info.as_ref().unwrap()
     }
+
+    fn raw_mutate(&mut self, data: &mut Vec<u8>) -> Result<()> {
+        // If a raw mutation function is configured then that's prioritized.
+        if let Some(mutate) = &self.raw_mutate_func {
+            return mutate(data);
```

Comment:
I think we need to pass a `max_size` parameter through to the raw mutation function. (a) `libfuzzer_sys::fuzzer_mutate` requires such a parameter, and (b) it lets us set `max_size == data.len() - 1` when we are configured to only do size-reducing mutations. When we aren't shrinking, then we can either pass a `max_size` of `data.capacity()` or `data.len() * 2` or something like that.